lib/SignalReceiver/src/TrafficHandler.h

```diff
@@ -0,0 +1,192 @@
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+
+/**
+ * The traffic handler class.
+ */
+class TrafficHandler
+{
+public:
+    /** The instance of Traffic handler. */
+    static TrafficHandler& getInstance()
+    {
+        static TrafficHandler instance;
+
+        return instance;
+    }
+
+    /**
+     * Process traffic.
+     *
+     * @returns true if processing is successful.
+     */
+    bool process();
+
+    /**
+     * Process list of IEs.
+     *
+     * @returns true if the IE list is successfully processed.
+     */
+    bool processList();
+
+    /**
+     * Add new IE to the list of traffic participants.
+     *
+     * @param[in] nameAsParameter is name of the IE
+     * @param[in] orientationAsParameter is orientation of IE
+     * @param[in] xPosAsParameter is position of IE on the x Axis
+     * @param[in] yPosAsParameter is position of IE on the y Axis
+     * @param[in] defaultDistanceAsParameter is set to 0 as default.
+     * @param[in] defaultPreviousDistanceAsParameter is set to 0 as default.
+     * @param[in] topicNameAsParameter is the MQTT topic to subscribe to.
+     */
+    bool setNewInfrastructureElement(const String& nameAsParameter, int32_t orientationAsParameter,
+                                     int32_t xPosAsParameter, int32_t yPosAsParameter,
+                                     int32_t defaultDistanceAsParameter, int32_t defaultPreviousDistanceAsParameter,
+                                     const String& topicNameAsParameter);
+
+    /**
+     * Set the value of received color.
+     *
+     * @param[in] receivedColor is the value of the color.
+     */
+    void setColorID(uint8_t receivedColor)
+    {
+        m_colorID = receivedColor;
+    }
+
+    /**
+     * Get the latest color ID.
+     *
+     * @returns the value of the color ID.
+     */
+    uint8_t getColorID()
+    {
+        return m_colorID;
+    }
+
+    /**
+     * Changes motor speeds based on the received color.
+     */
+    void processColor();
+
+    /**
+     * Check if robot-IE status is LOCKED_IN.
+     *
+     * @returns true if robot is locked onto IE.
+     */
+    bool checkLockIn();
+
+    /**
+     * Check if robot-IE status is NEAR.
+     *
+     * @returns true if robot is near the IE.
+     */
+    bool isNear();
+
+    /**
+     * Get the name of locked-onto IE.
+     *
+     * @returns the name of locked onto IE.
+     */
+    String getTargetName()
+    {
+        return lockedOnto;
+    }
+
+    /**
+     * Clears the target name.
+     */
+    void clearTarget()
+    {
+        lockedOnto = "";
+    }
+
+    /** Traffic handler constructor. */
+    TrafficHandler() : m_colorID(0U)

```

Comment:
Constructor and destructor must be private if you are working with a singleton

Reply:
Fixed